pylint/checkers/base/basic_error_checker.py

```diff
@@ -78,6 +78,36 @@ def _has_abstract_methods(node: nodes.ClassDef) -> bool:
     return len(utils.unimplemented_abstract_methods(node)) > 0
 
 
+def _new_instantiates_super(node: nodes.ClassDef) -> bool:
+    """Check if node implements `__new__`.
+
+    If `__new__` is implemented, check if it calls `super().__new__(cls)`.
+    """
+    if "__new__" not in node.locals:
+        return False
+
+    new = next(node.igetattr("__new__"))
+
+    calls = new.nodes_of_class(
+        nodes.Call, skip_klass=(nodes.FunctionDef, nodes.ClassDef)
+    )
+    for call in calls:
+        if not isinstance(call.func, nodes.Attribute):
+            continue
+        if call.func.attrname != "__new__":
+            continue
+
+        super_call = utils.safe_infer(call.func.expr)
+        if not isinstance(super_call, astroid.objects.Super):
+            continue
+
+        # `__new__` calls `super().__new__(cls)`.
+        return False
+
+    # If we get to this point, it means `__new__` has no calls to `super().__new__(cls)`.
+    return True
```

Comment:
This is the other way around from the docstring right? The function should probably be renamed.